pages/[league]/player/[id].tsx

```diff
@@ -0,0 +1,337 @@
+
+interface Props {
+  league: string;
+  id: number;
+}
+function PlayerPage({ league, id }: Props): JSX.Element {
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isSkater, setIsSkater] = useState<boolean>(true);
+  const [playerError, setPlayerError] = useState<boolean>(false);
+  const [filterSeasonType, setFilterSeasonType] = useState('Regular Season');
+
+  const router = useRouter();
+
+  // player info
+  const [playerName, setPlayerName] = useState('');
+  const [playerPosition, setPlayerPosition] = useState('');
+  const [playerTeam, setPlayerTeam] = useState('');
+  const [playerWeight, setPlayerWeight] = useState('');
+  const [playerHeight, setPlayerHeight] = useState('');
+
+  // ratings
+  const {
+    ratings: goalieRatings,
+    isLoading: isLoadingGoalieRating,
+    isError: isErrorGoalieRating,
+  } = useGoalieRatingsId(id, league);
+
+  const {
+    ratings: skaterRatings,
+    isLoading: isLoadingPlayerRating,
+    isError: isErrorPlayerRating,
+  } = useRatingsId(id, league);
+
+  // stats
+  const {
+    ratings: goalieStats,
+    isLoading: isLoadingGoalieStats,
+    isError: isErrorGoalieStats,
+  } = useGoalieStatsId(id, league, filterSeasonType);
+
+  const {
+    ratings: skaterStats,
+    isLoading: isLoadingSkaterStats,
+    isError: isErrorSkaterStats,
+  } = useSkaterStatsId(id, league, filterSeasonType);
+
+  // player info
+  const {
+    ratings: goalieInfo,
+    isLoading: isLoadingGoalieInfo,
+    isError: isErrorGoalieInfo,
+  } = useGoalieInfo(id, league)
+
+  const {
+    ratings: skaterInfo,
+    isLoading: isLoadingSkaterInfo,
+    isError: isErrorSkaterInfo,
+  } = useSkaterInfo(id, league)
+
+  // wait for all loads to complete
+  useEffect(() => {
+    setIsLoading(isLoadingGoalieRating || isLoadingPlayerRating || isLoadingGoalieStats
+      || isLoadingSkaterStats || isLoadingGoalieInfo || isLoadingSkaterInfo);
+    if (!isLoading) {
+      if (isErrorGoalieRating || isErrorPlayerRating || isErrorGoalieStats
+        || isErrorSkaterStats || isErrorSkaterInfo || isErrorGoalieInfo) {
+        setPlayerError(true);
+      } else {
+        if (skaterStats !== undefined && skaterInfo !== undefined) {
+          if (skaterStats.length > 0) {
+            setIsSkater(true);
+            setPlayerName(skaterStats[0].name);
+            setPlayerPosition(skaterStats[0].position);
+            setPlayerTeam(skaterStats[0].team.toString());
+            setPlayerHeight(skaterInfo[0].height.toString());
+            setPlayerWeight(skaterInfo[0].weight.toString());
+          } else {
+            setIsSkater(false);
+            setPlayerName(goalieStats[0].name);
+            setPlayerPosition('G');
+            setPlayerTeam(goalieStats[0].team.toString());
+            setPlayerHeight(goalieInfo[0].height.toString());
+            setPlayerWeight(goalieInfo[0].weight.toString());
+          }
+        }
+      }
+    }
+  });
```

Comment:
You should include a list of dependencies here so that we re-run this useEffect ONLY when things used here change.

Reply:
added